packages/db/schema.ts

```diff
@@ -55,101 +56,99 @@ export const discordVerificationStatus = pgEnum("discord_status", [
 	"rejected",
 ]);
 
-export const users = pgTable("users", {
-	clerkID: varchar("clerk_id", { length: 255 })
-		.notNull()
-		.primaryKey()
-		.unique(),
+export const userCommonData = pgTable("user_common_data", {
+	// id
+	clerkID: varchar("clerk_id", { length: 255 }).primaryKey(),
+
+	// data
 	firstName: varchar("first_name", { length: 50 }).notNull(),
 	lastName: varchar("last_name", { length: 50 }).notNull(),
 	email: varchar("email", { length: 255 }).notNull().unique(),
 	hackerTag: varchar("hacker_tag", { length: 50 }).notNull().unique(),
-	registrationComplete: boolean("registration_complete")
-		.notNull()
-		.default(false),
-	createdAt: timestamp("created_at").notNull().defaultNow(),
-	hasSearchableProfile: boolean("has_searchable_profile")
-		.notNull()
-		.default(true),
-	group: integer("group").notNull(),
+	age: integer("age").notNull(),
+	gender: varchar("gender", { length: 50 }).notNull(),
+	race: varchar("race", { length: 75 }).notNull(),
+	ethnicity: varchar("ethnicity", { length: 50 }).notNull(),
+	shirtSize: varchar("shirt_size", { length: 5 }).notNull(),
+	dietRestrictions: json("diet_restrictions").notNull(),
+	accommodationNote: text("accommodation_note"),
+	discord: varchar("discord", { length: 60 }).notNull(),
+	pronouns: varchar("pronouns", { length: 20 }).notNull(),
+	bio: text("bio").notNull(),
+	skills: json("skills").notNull().$type<string[]>().default([]),
+	profilePhoto: varchar("profile_photo", { length: 255 }).notNull(),
+
+	// metadata
+	isFullyRegistered: boolean("is_fully_registered").notNull().default(false),
+	signupTime: timestamp("signup_time").notNull().defaultNow(),
+	isSearchable: boolean("is_searchable").notNull().default(true),
 	role: roles("role").notNull().default("hacker"),
 	checkinTimestamp: timestamp("checkin_timestamp"),
-	teamID: varchar("team_id", { length: 50 }),
-	points: integer("points").notNull().default(0),
-	checkedIn: boolean("checked_in").notNull().default(false),
-	rsvp: boolean("rsvp").notNull().default(false),
-	approved: boolean("approved").notNull().default(false),
+	isRSVPed: boolean("is_rsvped").notNull().default(false),
+	isApproved: boolean("is_approved").notNull().default(false),
 });
 
-export const userRelations = relations(users, ({ one, many }) => ({
-	registrationData: one(registrationData, {
-		fields: [users.clerkID],
-		references: [registrationData.clerkID],
-	}),
-	discordVerification: one(discordVerification, {
-		fields: [users.clerkID],
-		references: [discordVerification.clerkID],
+export const userCommonRelations = relations(
+	userCommonData,
+	({ one, many }) => ({
+		hackerData: one(userHackerData, {
+			fields: [userCommonData.clerkID],
+			references: [userHackerData.clerkID],
+		}),
+		discordVerification: one(discordVerification, {
+			fields: [userCommonData.clerkID],
+			references: [discordVerification.clerkID],
+		}),
+		files: many(files),
+		scans: many(scans),
+		invites: many(invites),
```

Comment:
Mentioned in a below comment, but I think that the hacker data should also have the invites relation as well since only hackers can form teams and not common users. However, I would be interested to hear @Lermatroid's take on it.